hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -3387,26 +3489,41 @@ public EtagChecksum getFileChecksum(Path f, final long length)
   @Retries.OnceTranslated
   public RemoteIterator<LocatedFileStatus> listFiles(Path f,
       boolean recursive) throws FileNotFoundException, IOException {
-    return innerListFiles(f, recursive,
-        new Listing.AcceptFilesOnly(qualify(f)));
+    return toLocatedFileStatusIterator(innerListFiles(f, recursive,
+        new Listing.AcceptFilesOnly(qualify(f))));
+  }
+
+  private static RemoteIterator<LocatedFileStatus> toLocatedFileStatusIterator(
```

Comment:
I'm not sure if there is a better way to do this.  I have RemoteIterator<S3LocatedFileStatus> but need to return RemoteIterator<LocatedFileStatus> from the public methods like listFiles() so I use this to do the conversion.